metron-platform/metron-common/src/main/java/org/apache/metron/common/configuration/ConfigurationsUtils.java

```diff
@@ -76,7 +80,7 @@ public static void writeProfilerConfigToZookeeper(byte[] config, CuratorFramewor
   }
 
   public static void writeSensorParserConfigToZookeeper(String sensorType, SensorParserConfig sensorParserConfig, String zookeeperUrl) throws Exception {
-    writeSensorParserConfigToZookeeper(sensorType, JSONUtils.INSTANCE.toJSON(sensorParserConfig), zookeeperUrl);
+    writeSensorParserConfigToZookeeper(sensorType, JSONUtils.INSTANCE.toJSONPretty(sensorParserConfig), zookeeperUrl);
```

Comment:
So, I think the test is failing because you are pretty printing here.  `GlobalConfigServiceImplTest` functions are expecting to write out `{}` and retrieve back literally `{}` and, yet, you are pretty printing, which would yield `{ }`.  I think what you're doing is fine, but you probably want to adjust the failing `GlobalConfigServiceImplTest` cases to expect `{ }` rather than `{}`.

Reply:
Yeah, I hastily read that stacktrace. This was indeed the problem.